### PR TITLE
Log Jaeger traces

### PR DESCRIPTION
**Pull request by jodh-intel**
Defined `KATA_TESTS_LOGDIR` in the common bash library so that tests can write log information to a known location.

Save the trace spans logged with Jaeger to a directory below KATA_TESTS_LOGDIR` so that if the test fails, the spans can be analysed. This will allow the saved trace spans to be attached to a failing PR as artifacts. If the test passes, the cached spans are deleted.

Fixes: #1593.
    
Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: tracing/tracing-test.sh [hunk 1]
```diff
@@ -26,6 +28,13 @@ jaeger_docker_container_name="jaeger"
 cleanup(){
 	stop_jaeger 2>/dev/null || true
 	.ci/configure_tracing_for_kata.sh disable
+
+	# The tests worked so remove the logs
+	if sudo [ -d "$TRACE_LOG_DIR" ] && [ "$TRACE_LOG_DIR" != "/" ]
+	then
+		info "Removing cached trace logs from $TRACE_LOG_DIR"
+		sudo rm -rf "$TRACE_LOG_DIR"
```
Comment (jcvenegas) [inline tracing/tracing-test.sh]: @jodh-intel  I am  not clear when the trace logs are keep , seems that they are removed all the time ?

Comment (jodh-intel) [reply tracing/tracing-test.sh]: Good catch @jcvenegas!! I've moved the deletion logic to "main" now.

**Comment by jodh-intel** [comment]
@grahamwhaley - metrics CI failure:

```
18:17:31 INFO: delete stale docker resource under /var/lib/docker
18:17:31 FATAL: command execution failed
18:17:31 java.io.EOFException
```

**Comment by jodh-intel** [comment]
@chavafg - I've restarted the initrd CI, which failed with:

```
19:20:07 not ok 1 Pod quota
19:20:07 # (in test file k8s-pod-quota.bats, line 29)
19:20:07 #   `kubectl wait --for=condition=Available deployment/${deployment_name}' failed
19:20:07 # INFO: k8s configured to use runtimeclass
19:20:07 # resourcequota/pod-quota created
19:20:07 #     pods: "2"
19:20:07 #     pods: "2"
19:20:07 # deployment.apps/deploymenttest created
19:20:07 # error: timed out waiting for the condition on deployments/deploymenttest
19:20:07 # resourcequota "pod-quota" deleted
19:20:07 # deployment.extensions "deploymenttest" deleted
19:20:07 Makefile:104: recipe for target 'kubernetes' failed
```

**Comment by grahamwhaley** [comment]
metrics: odd fail - looks like it failed to do a cleanup of the docker dirs, which makes me feel maybe something got 'stuck' and was holding a file open. We reboot the metrics machine between (roughly) builds, so that should self-clear (and that is precisely why we do the reboot etc....). The build history on the metrics job looks pretty good/stable - I'll nudge a rebuild.

**Pull request merged**